feat: Add insider-threats framework (DPRK IT Workers) (#193)
security-alliance/frameworks: src/insider-threats/mitigating-dprk-it-workers.md

---
tags:
  - Security Specialist
  - Operations & Strategy
  - Community & Marketing
  - HR
  - Engineer/Developer
contributors:
- role: wrote
  users: [blackbigswan]
---

# Mitigating DPRK IT Workers

> This section discusses ways you can harden your organization against DPRK IT Workers, both before and after a potential hiring. All of the previously mentioned strategies apply but serve a more 'active' role. Here, we will discuss strategies that limit the effects of a DPRK IT Worker infiltration and what you should do if an infiltration is successful.

Reviewer: ```suggestion
This section discusses ways you can harden your organization against DPRK IT Workers, both before and after a potential hiring. All of the previously mentioned strategies apply but serve a more 'active' role to identify DPRK IT Workers during recruiting or within your organization. Here, we will discuss mitigation strategies that limit the effects of a DPRK IT Worker infiltration and what you should do after identifying a successful infiltration.
```

#### Hardening your hiring processes

> 1.  Apply all the steps from the sections "Am I Interviewing a DPRK IT Worker?" and "Did I Hire a DPRK IT Worker?". **It's important to educate all of your non-DPRK employees on these points, especially people responsible for hiring, developer relations, talent hunting, and community management.**

Reviewer 2: can link to those sections

2.  Introduce a culture of background checks. Do not take a prospective developer's claims at face value. **Even the most basic OSINT check can often discover deep inconsistencies. Check GitHub commit history, check Twitter history, and Google the full name of the developer.**
    1.  Is the work experience claimed in the CV reflected on GitHub?
    2.  Does the potential employee indicate any physical presence anywhere?
    3.  Is the potential employee writing non-AI-generated posts and replying to anything other than job offers on Twitter?
    4.  Did you find any name collisions after searching for the potential employee's name in Google?
    5.  Did you check the potential employee's nickname in Google? Maybe they have already appeared as a threat actor in a past report?
    6.  **Remember the best discovery rule:** You're looking for misrepresentations, mismatches, and obfuscation, not for North Koreans specifically. Tick your boxes on the first three and only then proceed to decide if you're potentially dealing with a North Korean.

> 3.  **Do not try to undercut developer salaries.** DPRK IT Workers will always accept low-ball offers without much hesitation. If you're looking for a cheap hire, you are at a higher chance of hiring a DPRK IT Worker.

Reviewer: ```suggestion
3.  **Do not try to undercut developer market salaries.** DPRK IT Workers will always accept low-ball offers without much hesitation. If you're looking for a cheap hire, you are at a higher chance of hiring a DPRK IT Worker.
```

5.  You could ask about North Korea on call and observe the reaction. There's a so-called "Fuck Kim Jong Un" test, where interviewer is asking to repeat the phrase and North Koreans are banned from saying it and they drop the call immediately. Obviously, it won't be effective for 'facilitators' but it will most likely be effective for North Koreans themselves. Do not rely on it as a panacea.
6.  Regardless of our tips of not focusing on "Asian man in his 20-30s" that particular appearance is still the most common (If not cloaked under the generative AI). Similarly, Korean-English accents.

#### Hardening your organization
1.  Define and implement tight access control rules for all of your employees, especially remote ones. Avoid giving dangerous permissions to workers who did not pass the DPRK discovery framework tests for 100% mark (all the points already discussed).
    1.  Do not give admin permissions to your GitHub organization and repositories, including for build releases, adding members, and repository settings.
    2.  Do not give publishing permissions on any package managers. The publishers of your packages are a critical supply chain risk; they need to be 100% vetted.

>     3.  Do not add your remote employees as multisignature wallet signers. The signer role should be reserved only for high-trust individuals, not based on the dedication of the developer you are working with.

Reviewer: ```suggestion
    3.  Do not add your remote employees as multisignature wallet signers. The signer role should be reserved for only the most trusted individuals in your organization or project.
```
Removed "dedication of developer" - doesn't strengthen the point imo

>     4.  Do not give unlimited access to cloud instances.

Reviewer: ```suggestion
    4.  Do not give unlimited access to cloud instances. Scope permissions to the minimal access required for each individual or team, including setting minimal expiration times for access tokens (e.g. access tokens should expire within hours, not weeks).
```

2.  Define a lead developer - a highly trusted individual you've met in real life and are sure of their background - as the code reviewer, publisher, signer, and image builder. This individual's role should be to double-check all of the remote workers' work and actions within your organization at all times. Make sure he is not North Korean.

> 3.  Silo your remote workers from the most critical infrastructure. If access to such a part of your infrastructure is to be given, always create a threat model for a 'worst-case' scenario.

Reviewer: ```suggestion
3.  Silo your remote workers from the most critical infrastructure. If you choose to provision access to such infrastructure, always create a threat model to understand the 'worst-case' scenario and potential mitigation approaches.
```

4.  Monitor your logs for IP addresses, user sessions, and timestamped access. Monitor your remote contributors' actions for any sudden changes (identity changes) and potential secrets leaks through their private accounts.
5.  **DO NOT allow hired contributors to use throwaway Github accounts for the work at your company!** It's part of the DPRK IT Worker tactics to save their main accounts from being discovered.

> 6.  Make all of your contributors to have at least *some* public code on Github in relation to your project. If you wall it, security researchers like SEAL911 won't be able to spot DPRK IT Workers for you and notify you privately. You may end up paying North Koreans for the whole duration of the project until you'll get hacked.

Reviewer 2: wait I'm a bit confused - could you add an example / clarify what `least *some* public code on Github in relation to your project`? Like they gotta push something to a public repo which is part of your org?

Author: Yeah, exactly. It may be as simple as about-us repository that lists all of the CODEOWNERS (contributors) to the organization with their Github usernames listed. Basically, we need to see who works in the organization, if all commits are private - it's harder.

#### I hired a DPRK IT Worker. What now?

> 1.  Contact security professionals if you're unable to handle the situation alone. You can reach out to SEAL_911.

Reviewer 2: SEAL911 naming is inconsistent. Paragraph above it was SEAL911, here is SEAL_911

Perhaps could reference @seal_911_bot on tg instead?

6.  Review the current and past work of the accounts in question. The code does not necessarily need to be malicious, but it can be of overall poor quality. **Pay extra attention to added dependencies or edited CI/CD files.**
7.  Review the permissions that these actors have or had in your organization. **Revoke them.**
8.  **Review all of the files and links previously provided by the DPRK IT Worker and who opened them.** The issue may extend beyond your technical workers. Look back to see if you did not potentially run malware-infested code or give them remote desktop access to your or your developers' machines.
9.  Set up a final confirmation call after all access is revoked and you are certain your organization is safe from any insider threats. **Attempt to perform KYC under some pretense. Do not settle for document scans, as these are easily obtainable.** Insist on a "video call" KYC, where the actor must show the physical document they previously provided next to their face.
10. Attempt to geolocate them. Send a document that tracks the IP address used to open it. Look for VPN usage or IP addresses in the DPRK/Eastern Russian range.

> 11. Collect all on-chain/payroll data on funds transferred to the actor. If an actor is confirmed as DPRK-related, you will be required to file a Suspicious Activity Report (SAR). We would appreciate you sharing this data with us later for public security reasons and future investigations. The data will remain private.

Reviewer 2: `We would appreciate you sharing this data with us` - maybe clarify you mean seal911 here?

Reviewer: Or better yet, share a specific means to share (TG bot, Discord, etc.)

Author: Adding link to securityalliance.org

#### Data collection
1.  You should collect the following data before firing the DPRK IT Worker. This data will help organizations like SEAL with future mitigation and the discovery of other actors.
    1.  Full Legal Name
    2.  Phone
    3.  Email(s)
    4.  Location (City/State)
    5.  LinkedIn
    6.  Telegram/Discord
    7.  GitHub
    8.  Copy of Resume
    9.  Date Hired
    10. Sourcing (Where were they hired from? Telegram, Staffing, Recruiter, etc.)
    11. Company Email (Yes/No)
    12. IP addresses (from email or any other sources at your disposal)
    13. Managed Laptop (Yes/No)
        1.  Shipment address (for the laptop or other items)
    14. KYC data (ID/Passport/Bills document scans)
    15. Photos/Screenshots (physical appearance, potentially from a video call)
    16. Payment data **(Extremely important!)**
        1.  On-chain addresses provided for payment
        2.  Bank data
    17. Did they refer anyone else?
    18. Were they referred by someone else?

> #### Overview of risks to your organization

Reviewer: I think this section would be better in README or General Information

Author: Yeah, I think you're right! Moving it there.

> #### Overview of risks to your organization

Reviewer: Suggest some others to add:
- Supply-chain compromise - DPRK IT Workers may intentionally introduce vulnerabilities that impact down-stream projects that depend on your software / services (e.g. SafeWallet UI in the ByBit hack)
- Reputational damage to your brand and loss of trust of your users and customers
- Asset freeze / loss of access to financial services: your assets may be frozen or seized, and financial institutions (e.g. banks, exchanges) may terminate your access if you are suspected of funding sanctioned entities
- Criminal investigations - law enforcement may investigate your involvement and impose fines or press criminal charges against your organization

1.  **Defrauding the company:** The company is paying someone whose identity they do not know.
2.  **Subpar operational security:** DPRK IT workers share credentials among themselves in open channels, have a poor command of Git, and unintentionally or intentionally leak the access they are granted to third parties.
3.  **Extortion:** They may try to extort more money after a job is finished.
4.  **Future hacking activities:** They may use the knowledge gained for future hacking activities.
5.  **Sanctions violations:** The DPRK is a sanctioned entity. No company can legally transfer funds to DPRK-related operations.

> 6.  **Contribution to the North Korean Military:** DPRK IT worker salaries directly contribute to the Military Ministry of North Korea. The workers do not keep the salaries for themselves.

Reviewer: Maybe this one is a sub-bullet of 5?
